src/neo-vm/ExecutionEngine.cs

```diff
@@ -1310,48 +1310,43 @@ private void ExecuteOp(OpCode opcode, ExecutionContext context)
                             break;
                         }
                     case OpCode.NEWARRAY:
+                    case OpCode.NEWSTRUCT:
                         {
-                            int count = (int)context.EvaluationStack.Pop().GetBigInteger();
+                            List<StackItem> items;
+                            var item = context.EvaluationStack.Pop();
 
-                            if (count < 0 || !CheckArraySize(count))
+                            if (item is VMArray array)
                             {
-                                State = VMState.FAULT;
-                                return;
-                            }
+                                // Allow to convert between array and struct
 
-                            List<StackItem> items = new List<StackItem>(count);
-                            for (var i = 0; i < count; i++)
-                            {
-                                items.Add(false);
+                                items = new List<StackItem>(array.Count);
+
+                                for (var i = 0; i < array.Count; i++)
+                                {
+                                    items.Add(array[i]);
```

Comment:
Given that `Struct` subclasses `Array`, I would think the fastest option could be to change the pointer of 
https://github.com/neo-project/neo-vm/blob/c552114fd8b513d9065021fa528a517a5e6ca2a6/src/neo-vm/Types/Array.cs#L10
to the equivalent of the new `Struct` object. The more items the more speed gain. I just don't know if that's possible with C#. Maybe with the `unsafe` keyword? It just feels bad seeing code looping over items to almost just change the type of the class. Just dropping ideas, no C# expert here.